flow/scripts/detail_route.tcl

```diff
@@ -1,4 +1,10 @@
 utl::set_metrics_stage "detailedroute__{}"
+if {[expr [file exists $::env(REPORTS_DIR)/congestion.rpt] && \
+    [file size $::env(REPORTS_DIR)/congestion.rpt] != 0]} {
+  error "Global routing failed, run `make gui_grt` and load $::env(REPORTS_DIR)/congestion.rpt \
```

Comment:
Detailed route does not proceed if global route had congestion failures.

Reply:
This moves the error to the wrong step.  We already have
```
set result [catch {do_global_route} errMsg]
if {$result != 0} {
  write_db $::env(RESULTS_DIR)/5_1_grt-failed.odb
  error $errMsg
}
```
Why isn't this sufficient?

Reply:
Separation of concerns and also trying to minimize surprises.

I want the responsiblity for providing .odb files and .rpt in the case of global/detailed routing failures and the detection of fatal(no artifacts should be produced) to reside inside ORFS.

The interface I want between ORFS and bazel is that ORFS provides .odb and .rpt file and a non-zero exit code, in which case artifacts are created. With a non-zero exit code, then no artifacts are created. A non-zero exit code is ORFS's way to tell bazel that there was a fatal error and no artifacts should be created.

This mirrors detailed routing's behavior today, I want the same for global routing.

Yes: this means that there needs to be a check not to continue detailed routing when there is a global routing error. This is somewhat surprising, so not ideal, but the error message in detailed routing is easily visible and not drowned out.

One could imagine the same for detailed routing; that there's no point in continuing with the final stage if there the DRC errors in detailed routing are unacceptable(though I don't know what "unacceptable" is for detailed routing).

Reply:
I dont think so: this is how detailed route works.

Reply:
It is how gpl & dpl work.  drt is different as people often manually fix drcs left by drt.

Reply:
does dpl and other stages have similarly useful reports in case of failure, other than the log?

Reply:
You can use -report_file_name with detailed_place though it looks like we don't currently in the flow.  I'm not sure why it matters.

Reply:
Just trying to understand where there are frequently useful reports on failure. More reports on place failures could be a useful future improvement.